Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -178,32 +179,29 @@ void SoftwareBreakpointManager::getOpcode(uint32_t type,
 }
 
 ErrorCode SoftwareBreakpointManager::enableLocation(Site const &site) {
-  std::string opcode;
-  std::string old;
+  ByteVector opcode;
+  ByteVector old;
   ErrorCode error;
 
   getOpcode(site.size, opcode);
   old.resize(opcode.size());
-  error = _process->readMemory(site.address, &old[0], old.size());
+  error = _process->readMemory(site.address, old.data(), old.size());
   if (error != kSuccess) {
     DS2LOG(Error, "cannot enable breakpoint at %#lx, readMemory failed",
            (unsigned long)site.address.value());
     return error;
   }
 
-  error = _process->writeMemory(site.address, &opcode[0], opcode.size());
+  error = _process->writeMemory(site.address, opcode.data(), opcode.size());
   if (error != kSuccess) {
     DS2LOG(Error, "cannot enable breakpoint at %#lx, writeMemory failed",
            (unsigned long)site.address.value());
     return error;
   }
 
-  DS2LOG(Debug, "set breakpoint instruction %#lx at %#lx (saved insn %#lx)",
-         (unsigned long)(site.size == 2 ? *(uint16_t *)&opcode[0]
-                                        : *(uint32_t *)&opcode[0]),
-         (unsigned long)site.address.value(),
-         (unsigned long)(site.size == 2 ? *(uint16_t *)&old[0]
-                                        : *(uint32_t *)&old[0]));
+  DS2LOG(Debug, "set breakpoint instruction %s at %#lx (saved insn %s)",
+         ToHex(opcode).c_str(), (unsigned long)site.address.value(),
```

Comment:
Not your code, but can you also use `PRI_PTR` and `PRI_PTR_CAST` for the address printing here?